node/src/neighborhood/overall_connection_status.rs

```diff
@@ -229,20 +226,30 @@ impl OverallConnectionStatus {
 
     pub fn update_ocs_stage_and_send_message_to_ui(
         &mut self,
+        new_stage: OverallConnectionStage,
         node_to_ui_recipient: &Recipient<NodeToUiMessage>,
+        logger: &Logger,
     ) {
-        // For now, this function is only called when Standard or Introduction Gossip
-        // is received, as it is implemented only for the advancing transitions right now
-        // TODO: Modify this fn when you're implementing the regressing transitions and try to
-        // write a more generalized fn, which can be called when any stage gets updated
+        // TODO: Modify this fn when you're implementing the regressing transitions
         let prev_stage = self.stage;
-        if self.can_make_routes() {
-            self.stage = OverallConnectionStage::ThreeHopsRouteFound;
-        } else {
-            self.stage = OverallConnectionStage::ConnectedToNeighbor;
-        }
-        if self.stage as usize > prev_stage as usize {
+        if new_stage as usize > prev_stage as usize {
+            self.stage = new_stage;
             OverallConnectionStatus::send_message_to_ui(self.stage.into(), node_to_ui_recipient);
+            debug!(
+                logger,
+                "The stage of OverallConnectionStatus has been changed \
+                from {:?} to {:?}. A message to the UI was also sent.",
+                prev_stage,
+                new_stage
+            );
+        } else {
+            trace!(
+                logger,
+                "There was an attempt to change the stage of OverallConnectionStatus \
```

Comment:
Maybe "regress" instead of "change" to make clear the reason why the attempt was discarded?